src/pages/workflowThemes/workflowThemes.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+// import Image from "next/image";
+import Box from "@mui/material/Box";
+import Card from "@mui/material/Card";
+import Typography from "@mui/material/Typography";
+import CardActionArea from "@mui/material/CardActionArea";
+import { CardHeader, CardMedia } from "@mui/material";
+import { useRouter } from "next/router";
+import { useWorkflows } from "@/context/DBContext";
+import { PreconfigContext, ThemesContext } from "@/models/workflow";
+
+export default function PreConfigWorkflows() {
+    const router = useRouter();
+    const [selectedCard, setSelectedCard] = React.useState<string>("");
+
+    const context = useWorkflows();
+    const themes = context?.themes;
+  
+    const themesId = router?.query?.id;
+    // Get theme info from theme query
+    const theme = themes?.find(
```

Comment:
Some of these lookups are pretty clunky. Is it possible to create a component that takes a `PreconfigContext` as a prop, and pass it to the component once you've checked that it's not null? That would reduce the amount you have to check for null.